opal/models.py

```diff
@@ -391,6 +393,49 @@ def get_active_episode(self):
                 return episode
         return None
 
+    @transaction.atomic()
+    def bulk_update(self, dict_of_list_of_upgrades, user, episode=None):
+        """
+                takes in a dictionary of api name to a list of fields and
+                creates the required subrecords. If passed an episode
+                sub record but no episode it will create an episode
+                and attatch it.
+
+                e.g. {"allergies": [
+                            {"drug": "paracetomol"}
+                            {"drug": "aspirin"}
+                          ],
+                      "investigation":[
+                            {"name": "some test", "details": "some details"}
+                          ]
+                     }
+        """
+        if "demographics" not in dict_of_list_of_upgrades:
+            if not self.id:
+                raise ValueError(
+                    "demographics are required when creating a new patient"
```

Comment:
...also, it's a method on an instantiated Patient - are you sure you want to always raise a valueerror if there's no demographics?